### PR TITLE
[CPU] fix coverity issue in multiclass and a bug in matrix_nms

### PR DESCRIPTION
**Pull request by tiger100256-hu**
### Details:
 - *fix coverity issue in  multiclass*
 - *fix a bug in matrix_nms*

### Tickets:
 - *108988*

### CODE REVIEW

**Comment by tiger100256-hu** [comment]
@ceciliapeng2011  help to review, thanks

File: src/plugins/intel_cpu/src/nodes/matrix_nms.cpp [hunk 1]
```diff
@@ -376,7 +376,7 @@ void MatrixNms::execute(dnnl::stream strm) {
 
     // NMS-alike nodes are always transformed to NMSIEInternal node in case of legacy api, for compatibility.
     // And on the other hand in case of api 2.0, keep them internal dynamic for better performance and functionality.
-    if (isDynamicNode()) {
+    if (!m_outStaticShape) {
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/nodes/matrix_nms.cpp:379]: Could you please elaborate on why I decided to change the condition?

Comment (tiger100256-hu) [reply src/plugins/intel_cpu/src/nodes/matrix_nms.cpp:379]: @ceciliapeng2011  could you help to explain it, thanks.
@maxnick  this condition change is from cecilia, I just submit it with the coverity issue of multiclass

Comment (ceciliapeng2011) [reply src/plugins/intel_cpu/src/nodes/matrix_nms.cpp:379]: @maxnick It is a leftover of my last PR. https://github.com/openvinotoolkit/openvino/pull/16128#discussion_r1148861999 It's got merged before I fix it. that's all.

**Review by ceciliapeng2011 — APPROVED** [review]

File: src/plugins/intel_cpu/src/nodes/matrix_nms.cpp (hunk 1 above)
Comment (maxnick) [reply src/plugins/intel_cpu/src/nodes/matrix_nms.cpp:379]: Thank you, now it is clear.

**Review by maxnick — APPROVED** [review]

**Pull request merged**